### PR TITLE
Feature/#777

### PR DESCRIPTION
**Pull request by schoicsiro**
Would you be able to get PR?

### CODE REVIEW

File: grails-app/controllers/au/org/ala/profile/api/ApiController.groovy [hunk 1]
```diff
@@ -34,6 +34,7 @@ class ApiController extends BaseController {
     MapService mapService
     ApiService apiService
 
+    @RequireApiKey()
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:37]: why is requireapikey required? Similarly other below actions.

File: grails-app/controllers/au/org/ala/profile/api/ApiController.groovy [hunk 2]
```diff
@@ -106,6 +107,41 @@ class ApiController extends BaseController {
         }
     }
 
+    @Path("/api/opus/list")
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:110]: rename to `/api/opus` for consistency with other APIs

File: grails-app/controllers/au/org/ala/profile/api/ApiController.groovy [hunk 3]
```diff
@@ -106,6 +107,41 @@ class ApiController extends BaseController {
...
+    @Operation(
+            summary = "Get all collection (opus) details",
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:112]: Perhaps better to reword it to `Get all public collections`

File: grails-app/controllers/au/org/ala/profile/api/ApiController.groovy [hunk 4]
```diff
@@ -106,6 +107,41 @@ class ApiController extends BaseController {
...
+                    @ApiResponse(responseCode = "405",
+                            description = "An unexpected error has occurred while processing your request."),
+                    @ApiResponse(responseCode = "404",
+                            description = "Collection not found"),
+                    @ApiResponse(responseCode = "500",
+                            description = "An unexpected error has occurred while processing your request.")
+            ]
+    )
+    def getListCollections () {
+       List opus = profileService.getOpusList() as List
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:140]: Reuse `profileService.getOpus()`. No need to implement a custom code in profile-hub or profile-service.

File: grails-app/controllers/au/org/ala/profile/api/ApiInterceptor.groovy [hunk 5]
```diff
@@ -26,6 +26,8 @@ class ApiInterceptor {
         Class controllerClass = controller?.clazz
         def method = controllerClass?.getMethod(actionName, [] as Class[])
 
+        if (method.name == "getListCollections") return true
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/ApiInterceptor.groovy:29]: should not escape api interceptor logic.

File: grails-app/services/au/org/ala/profile/hub/ProfileService.groovy [hunk 6]
```diff
@@ -45,6 +45,11 @@ class ProfileService {
         webServiceWrapperService.get("${grailsApplication.config.profile.service.url}/opus/${encPath(opusId)}", [:], ContentType.APPLICATION_JSON, true, false, getCustomHeaderWithUserId())?.resp
     }
 
+    def getOpusList() {
```
Comment (temi) [inline grails-app/services/au/org/ala/profile/hub/ProfileService.groovy:48]: See comment above.

File: grails-app/controllers/au/org/ala/profile/hub/UrlMappings.groovy [hunk 7]
```diff
@@ -263,6 +263,7 @@ class UrlMappings {
             get "/opus/$opusId/profile/$profileId/image" (version: "1.0", controller: "api", action: "getImages", namespace: "v1")
             get "/opus/$opusId/profile/$profileId/attribute/$attributeId" (version: "1.0", controller: "api", action: "getAttributes", namespace: "v1")
             get "/opus/$opusId/profile/$profileId/draft" (version: "1.0", controller: "api", action: "getDraftProfile", namespace: "v1")
+            get "/opus/list" (version: "1.0", controller: "api", action: "getListCollections", namespace: "v1")
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/hub/UrlMappings.groovy:266]: should be `/api/opus`

File: grails-app/controllers/au/org/ala/profile/api/ApiController.groovy [hunk 8]
```diff
@@ -333,6 +370,7 @@ class ApiController extends BaseController {
         }
     }
 
+    @RequireApiKey()
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:373]: Remove. applies to other implementations.

**Review by temi — CHANGES_REQUESTED** [review]

File: grails-app/controllers/au/org/ala/profile/api/ApiController.groovy (hunk 1 above)
Comment (schoicsiro) [reply grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:37]: previous ApiController class had @RequireApiKey. so I put it into all method except api/opus/list

Comment (temi) [reply grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:37]: Keep security standard for all actions.

File: grails-app/controllers/au/org/ala/profile/domain/CollectionList.groovy [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package au.org.ala.profile.domain
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/domain/CollectionList.groovy:1]: Move this class to `src/main/groovy/au/org/ala/profile/api/`.

File: grails-app/controllers/au/org/ala/profile/api/ApiController.groovy [hunk 10]
```diff
@@ -106,6 +109,43 @@ class ApiController extends BaseController {
...
+            operationId = "/api/opus",
+            method = "GET",
+            responses = [
+                    @ApiResponse(
+                            responseCode = "200",
+                            content = @Content(
+                                    mediaType = "application/json",
+                                    array = @ArraySchema(
+                                            schema = @Schema(
+                                                    implementation = OpusResponse.class
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/ApiController.groovy:125]: Replace with CollectionList since OpusResponse has a lost more properties than what is returned by this API.

**Review by temi — CHANGES_REQUESTED** [review]

File: grails-app/controllers/au/org/ala/profile/api/CollectionList.groovy [hunk 11]
```diff
@@ -1,4 +1,4 @@
-package au.org.ala.profile.domain
+package au.org.ala.profile.api
```
Comment (temi) [inline grails-app/controllers/au/org/ala/profile/api/CollectionList.groovy:1]: The location is not correct. Should not place this file with controllers. See comment before to get exact location.

**Review by temi — CHANGES_REQUESTED** [review]

**Review by temi — APPROVED** [review]
